C86 compiler assert on shift operation

This one
```
int main()
{
	unsigned int i, j;
	unsigned char k, l;

	l = i << (1 - (j << k));

}
```
```
make -f Makefile.c86 test
/home/rafael2k/programs/devel/8086-toolchain/host-bin/cpp86 -0  -I/home/rafael2k/programs/devel/elks/libc/include -I/home/rafael2k/programs/devel/elks/elks/include -I/home/rafael2k/programs/devel/elks/libc/include/c86 -D__LIBC__ -D__HAS_NO_FLOATS__=1 -D__HAS_NO_LONGLONG__ -o test.i test.c
/home/rafael2k/programs/devel/8086-toolchain/host-bin/c86 -g -O -bas86 -warn=4 -lang=c99 -align=yes -separate=yes -stackopt=minimum -peep=all -stackcheck=no  test.i test.as
c86: regx86.c:168: g_push: Assertion `0' failed.
TERMINATING: - signal 6 received

make: *** [Makefile.c86:59: test.as] Error 1
rm test.i test.as
```

--- a/compiler/regx86.c
+++ b/compiler/regx86.c
@@ -150,8 +150,9 @@ static void g_push P2 (REG, reg, DEEP, depth)
     sync_stack ();
     ap = mk_reg (reg);
     switch (reg_alloc[depth].regtype) {
-    case D_REG | T_REG:
     case A_REG | T_REG:
+    case D_REG | T_REG:
+    case D_REG | T_REG | N_REG: /* ghaerr fix shift expression bug by allowing no CX */
     case X_REG | T_REG:
 	g_code (op_push, small_option ? IL2 : IL4, ap, NIL_ADDRESS);
 	break;
@@ -214,6 +215,7 @@ static void g_pop P2 (REG, reg, DEEP, depth)
     switch (reg_alloc[depth].regtype) {
     case A_REG | T_REG:
     case D_REG | T_REG:
+    case D_REG | T_REG | N_REG: /* ghaerr fix shift expression bug by allowing no CX */
     case X_REG | T_REG:
 	g_code (op_pop, small_option ? IL2 : IL4, ap, NIL_ADDRESS);
 	break;

